From 0375215beb18805fb769ede4769e66109fd37518 Mon Sep 17 00:00:00 2001
From: Mahesh R Vaidya <mahesh.r.vaidya@intel.com>
Date: Mon, 19 May 2025 20:48:13 +0530
Subject: [PATCH] Workaround patch for 1G crc error

When HPS EMAC is set to GMII 1G mode, Clock Muxes in EMAC top
wrapper selects a different clock source compared to other data 
rates (2.5G, 100Mbps, 10Mbps) in GMII mode leading to Data loss 
in TX direction due to clock PPM variations when Multi-rate PHY 
is used. This is a workaround patch to ensure the clock source 
selected for 1G is the same as that of the other data rates.  

Upstream-Status: Inappropriate (Workaround patch)
Signed-off-by: Mahesh R Vaidya <mahesh.r.vaidya@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/dwxgmac2_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwxgmac2_core.c b/drivers/net/ethernet/stmicro/stmmac/dwxgmac2_core.c
index 0d850ad4af39..9185422a15a6 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwxgmac2_core.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwxgmac2_core.c
@@ -2042,7 +2042,7 @@ int dwxgmac2_setup(struct stmmac_priv *priv)
 	mac->link.duplex = 0;
 	mac->link.speed10 = XGMAC_CONFIG_SS_10_MII;
 	mac->link.speed100 = XGMAC_CONFIG_SS_100_MII;
-	mac->link.speed1000 = XGMAC_CONFIG_SS_1000_GMII;
+	mac->link.speed1000 = XGMAC_CONFIG_SS_2500_GMII;
 	mac->link.speed2500 = XGMAC_CONFIG_SS_2500_GMII;
 	mac->link.xgmii.speed2500 = XGMAC_CONFIG_SS_2500;
 	mac->link.xgmii.speed5000 = XGMAC_CONFIG_SS_5000;
-- 
2.34.1

